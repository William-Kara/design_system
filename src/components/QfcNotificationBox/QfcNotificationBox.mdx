import { ArgTypes, Meta } from '@storybook/blocks';
import QfcNotificationBox from '@/components/QfcNotificationBox/QfcNotificationBox.vue';

<Meta title="Modules/Notifications/NotificationBox" />

# Definition & Usage

## Props

The following props are available for the QfcSidebar component:

<ArgTypes of={QfcNotificationBox} />

To use QfcNotificationBox, style needs to be imported separately.

```sass
@import '@website/design-system-vuejs/dist/styles/components/QfcBox.scss';
@import '@website/design-system-vuejs/dist/styles/components/QfcNotificationBox.scss';
```

```html
<template>
  <QfcNotificationBox type="success" />
</template>
```
