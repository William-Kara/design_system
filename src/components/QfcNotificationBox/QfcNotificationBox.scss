$edge-offset: $q-generic-space-1;
$parent-border-radius: $br-m;

.qfc-notification-box {
  display: grid;
  grid-template-columns: auto 1fr;
  align-items: center;

  &__edge {
    width: $edge-offset;
    margin-right: -$edge-offset;
    height: calc(100% - 1.8*$parent-border-radius );
    border-radius: 0 1rem 1rem 0;
    z-index: 1;
  }
  &__content-container {
    flex: 1;
  }
}

.--info {
  background-color: $info;
}
.--success {
  background-color: $success;
}
.--warning {
  background-color: $warning;
}
.--danger {
  background-color: $danger;
}
.--none {
  background-color: transparent;
}
