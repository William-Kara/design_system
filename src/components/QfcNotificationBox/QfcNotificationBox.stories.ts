import NotificationBox from './QfcNotificationBox.vue';
import type { Meta, StoryObj } from '@storybook/vue3';

const meta: Meta<typeof NotificationBox> = {
  component: NotificationBox,
  title: 'Modules/Notifications/NotificationBox',
  argTypes: {
    type: {
      options: ['success', 'danger', 'warning', 'info', 'none'],
    },
  },
};
export default meta;

type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    type: 'success',
  },
};
