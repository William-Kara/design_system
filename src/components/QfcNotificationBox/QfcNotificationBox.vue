<template>
  <div class="qfc-notification-box">
    <div class="qfc-notification-box__edge" :class="notificationColorStyle"></div>
    <QfcBox padding="small" class="qfc-notification-box__content-container">
      <slot />
    </QfcBox>
  </div>
</template>

<script lang="ts" setup>
import QfcBox from '@/components/QfcBox/QfcBox.vue';
import { computed } from 'vue';

type NotificationType = 'success' | 'danger' | 'warning' | 'info' | 'none';
interface Props {
  type: NotificationType;
}
const props = withDefaults(defineProps<Props>(), {
  type: 'none',
});

const notificationColorStyle = computed(() => `--${props.type}`);
</script>

<style lang="scss" scoped src="./QfcNotificationBox.scss" />
