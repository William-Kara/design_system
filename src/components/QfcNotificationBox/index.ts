import QfcNotificationBox from './QfcNotificationBox.vue';

export { QfcNotificationBox };
export default QfcNotificationBox;
