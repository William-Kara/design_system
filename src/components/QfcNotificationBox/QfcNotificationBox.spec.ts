import { it, describe, beforeEach, expect } from 'vitest';

import { shallowMount } from '@vue/test-utils';
import QfcNotificationBox from './QfcNotificationBox.vue';

let wrapper: any = null;

const propsValues = {
  type: 'success',
};

beforeEach(() => {
  wrapper = shallowMount(QfcNotificationBox, {
    propsData: propsValues,
  });
});

describe('QfcNotificationBox Component', () => {
  it('should mount properly', () => {
    expect(wrapper).toBeTruthy();
  });

  it('should have correct props', () => {
    expect(wrapper.props()).toEqual(propsValues);
  });

  it('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
