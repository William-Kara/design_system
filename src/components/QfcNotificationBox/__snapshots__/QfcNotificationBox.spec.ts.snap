// Vitest Snapshot v1

exports[`QfcNotificationBox Component > should render correctly 1`] = `
"<div data-v-271437d3=\\"\\" class=\\"qfc-notification-box\\">
  <div data-v-271437d3=\\"\\" class=\\"qfc-notification-box__edge --success\\"></div>
  <qfc-box-stub data-v-271437d3=\\"\\" size=\\"fullwidth\\" padding=\\"small\\" class=\\"qfc-notification-box__content-container\\"></qfc-box-stub>
</div>"
`;
